sklearn/ensemble/_stacking.py

```diff
@@ -72,6 +81,11 @@ def _concatenate_predictions(self, predictions):
                     X_meta.append(preds[:, 1:])
                 else:
                     X_meta.append(preds)
+        if self.passthrough:
+            X_meta.append(X)
+            if sparse.issparse(X):
+                return sparse.hstack(X_meta).tocsr()
```

Comment:
More simplifications Nit: 

```py
if self.passthrough:
    X_meta.append(X)

hstack = sparse.hstack if sparse.issparse(X) else np.hstack
return hstack(X_meta)
```

Reply:
I tried a few ways to get this to work but kept running into errors since `sparse.hstack` returns a `coo_matrix` instead of a `csr_matrix`.

Is there a clever way to embed the `.tocsr()` function within the `if/else` line? I didn't want to add it as `return hstack(X_meta).tocsr()` since it would try to force dense matrices into `csr` format also.

Reply:
should this be `toformat(X.format)`??

Reply:
Not sure what you mean by embedding .tocsr in the if/else line.

(I am surprised that `sparse.hstack` works with a mix of dense and sparse input... I'm sure the docs only mention sparse.)

Reply:
> (I am surprised that sparse.hstack works with a mix of dense and sparse input... I'm sure the docs only mention sparse.)

It is surprising. I am unsure if we should rely on this behavior.

@jcusick13 Maybe something like this will work:

```py
if self.passthrough:
    X_meta.append(X)

if sparse.issparse(X):
    return sparse.hstack(X_meta, format=X.format)

return np.hstack(X_meta)
```

Reply:
Good catch on using `X.format` instead of hardcoding `csr`.

The `sparse.hstack` documentation is a bit worrying with the explicit mention of sparse inputs (and the sparse statements are just as explicit in the [`sparse.bmat` docs](https://docs.scipy.org/doc/scipy/reference/generated/scipy.sparse.bmat.html), which is called directly by `sparse.hstack`).

I took a look at the `sparse.bmat` [source](https://github.com/scipy/scipy/blob/v1.3.1/scipy/sparse/construct.py#L570) and it looks like it's converting all of its inputs into a COO matrix in a way that would work regardless of sparse vs. dense input (by just using `sparse.coo_matrix()`). I ran some simple tests locally and it never seemed to raise a problem --
```
sparse.hstack([np.ones(10), sparse.csr_matrix(np.ones(10))])
>>> <1x20 sparse matrix of type '<class 'numpy.float64'>'
	with 20 stored elements in COOrdinate format>
```
The current setup was passing all tests locally but definitely open to hearing your guys' thoughts on the matter.